sky/cli.py

```diff
@@ -638,73 +639,124 @@ def _add_options(func):
     return _add_options
 
 
-def _parse_override_params(
-        cloud: Optional[str] = None,
-        region: Optional[str] = None,
-        zone: Optional[str] = None,
-        gpus: Optional[str] = None,
-        cpus: Optional[str] = None,
-        memory: Optional[str] = None,
-        instance_type: Optional[str] = None,
-        use_spot: Optional[bool] = None,
-        image_id: Optional[str] = None,
-        disk_size: Optional[int] = None,
-        disk_tier: Optional[str] = None,
-        ports: Optional[Tuple[str]] = None) -> Dict[str, Any]:
-    """Parses the override parameters into a dictionary."""
-    override_params: Dict[str, Any] = {}
+def _parse_resources_override_params(
+    cloud: Optional[str] = None,
+    region: Optional[str] = None,
+    zone: Optional[str] = None,
+    gpus: Optional[str] = None,
+    cpus: Optional[str] = None,
+    memory: Optional[str] = None,
+    instance_type: Optional[str] = None,
+    use_spot: Optional[bool] = None,
+    image_id: Optional[str] = None,
+    disk_size: Optional[int] = None,
+    disk_tier: Optional[str] = None,
+    ports: Optional[Tuple[str]] = None,
+) -> Dict[str, Any]:
+    """Parses the resources override parameters into a dictionary."""
+    resources_override_params: Dict[str, Any] = {}
     if cloud is not None:
         if cloud.lower() == 'none':
-            override_params['cloud'] = None
+            resources_override_params['cloud'] = None
         else:
-            override_params['cloud'] = clouds.CLOUD_REGISTRY.from_str(cloud)
+            resources_override_params['cloud'] = (
+                clouds.CLOUD_REGISTRY.from_str(cloud))
     if region is not None:
```

Comment:
I am wondering if we can refactor out this common pattern
```
    if field is not None:
         if field.lower() == 'none':
             resources_override_params['field_str'] = None
         else:
             resources_override_params['field_str'] = field
```

Reply:
This is a little bit tricky since some fields like `clouds` need special handling; Lets do this in another PR?

Reply:
sg.